snooty/postprocess.py

```diff
@@ -812,67 +850,141 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
         If not, save this node to be populated at the end of the page.
         """
 
-        try:
-            line = node.span[0]
-            if isinstance(node, n.SubstitutionDefinition):
-                self.substitution_definitions[node.name] = node.children
-                self.seen_definitions = set()
-            elif isinstance(node, n.SubstitutionReference):
-                # Get substitution from page. If not found, attempt to source from snooty.toml. Otherwise, save substitution to be populated at the end of page
-                substitution = self.substitution_definitions.get(
-                    node.name
-                ) or self.project_config.substitution_nodes.get(node.name)
-
-                if (
-                    self.seen_definitions is not None
-                    and node.name in self.seen_definitions
-                ):
-                    # Catch circular substitution
-                    del self.substitution_definitions[node.name]
-                    node.children = []
-                    self.context.diagnostics[fileid_stack.current].append(
-                        SubstitutionRefError(
-                            f'Circular substitution definition referenced: "{node.name}"',
-                            line,
-                        )
-                    )
-                elif substitution is not None:
-                    node.children = substitution
-                else:
-                    # Save node in order to populate it at the end of the page
-                    self.unreplaced_nodes.append((node, line))
-
-                if self.seen_definitions is not None:
-                    self.seen_definitions.add(node.name)
-        except KeyError:
-            # If node does not contain "name" field, it is a duplicate substitution definition.
-            # An error has already been thrown for this on parse, so pass.
-            pass
+        if isinstance(node, n.Directive):
+            if node.name != "include":
```

Comment:
Will this be updated when the sharedinclude directive gets added?

Reply:
Yepyep!